### ISSUE TITLE
MD037/no-space-in-emphasis reported with formulas

### ISSUE THREAD

**Issue by stefan-muc** [DavidAnson/markdownlint#336]
MD037/no-space-in-emphasis gets reported on first list element:

```
# Test Document

$$foobar$$ (1)

This is some rubbish text that can also be left out

* foobar
* foobar line that can be left out
* $foobar$
```

Which will render as:
![Clipboard01](https://user-images.githubusercontent.com/544335/94936468-908f0c00-04ce-11eb-817e-2b0398f76cbf.png)

I'm using [markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) in Visual Studio Code together with [Markdown+Math](https://marketplace.visualstudio.com/items?itemName=goessner.mdmath). Last one uses $ and $$ for inline formulas just like LATEX does it.
Markdownlit seems to be confused by it, because it reports fist list element as "MD037/no-space-in-emphasis" when surrounded by

* a full line formula before that list
* this formula having an formula number in brackets
* an inline formula later on in that document (not necessarily in that list)

**Comment by DavidAnson** [comment]
The `markdownlint` library doesn't report errors for this example, so the issue may be with how the https://www.npmjs.com/package/@iktakahiro/markdown-it-katex plugin used by the VS Code extension is behaving. I'll have a look.

https://dlaa.me/markdownlint/#%25m%23%20Test%20Document%0A%0A%24%24foobar%24%24%20(1)%0A%0AThis%20is%20some%20rubbish%20text%20that%20can%20also%20be%20left%20out%0A%0A*%20foobar%0A*%20foobar%20line%20that%20can%20be%20left%20out%0A*%20%24foobar%24%0A
